### PR TITLE
introduce OpenTracing

### PR DESCRIPTION
**Pull request by alexkorotkikh**
Issue #14

### CODE REVIEW

**Comment by vibhory2j** [comment]
👍

**Comment by alexkorotkikh** [comment]
@mohabusama please take a look. 

I've decided not to remove `preHandle` and `afterCompletion` logs from the span, because in general, it's possible only together with disabling all Spring-related auto-instrumentation, so that even such basic things like `http.method` tags have to be added manually.

**Comment by alexkorotkikh** [comment]
Hi @mohabusama, can we merge this?

File: src/main/java/de/zalando/zmon/metriccache/Application.java [hunk 1]
```diff
@@ -49,122 +57,180 @@
...
-
-            URIBuilder builder = new URIBuilder();
-            URI uri = builder.setScheme("http").setHost(targetHost).setPort(Integer.parseInt(config.getServer_port())).setPath("/api/v1/rest-api-metrics/kairosdb-format").setParameter("redirect", "false")
-                    .setParameter("application_id", applicationId)
-                    .setParameter("application_version", applicationVersion).build();
-
-            String body = executor.execute(Request.Get(uri)).returnContent().asString();
-            response.setContentType(ContentType.APPLICATION_JSON.getMimeType());
-            writer.write(body);
+        } catch (IOException ex) {
```
Comment (mohabusama) [inline src/main/java/de/zalando/zmon/metriccache/Application.java:146]: We should add the `error=true` tag and may be log the exception in the span

Comment (alexkorotkikh) [reply src/main/java/de/zalando/zmon/metriccache/Application.java:146]: opentracing already takes case of this: https://github.com/opentracing-contrib/java-web-servlet-filter/blob/master/opentracing-web-servlet-filter/src/main/java/io/opentracing/contrib/web/servlet/filter/ServletFilterSpanDecorator.java#L106

**Review by mohabusama — APPROVED** [review]

**Comment by mohabusama** [comment]
👍

**Comment by alexkorotkikh** [comment]
:+1:

**Pull request merged**